Support for Figma on Windows

snwagbata opened the pull request.

snwagbata committed:
prevent shell scripts from executing on windows

snwagbata committed:
prevent 'test' as a projectName; convertFile() parameter renamed

snwagbata committed:
modified output test files to run on windows

snwagbata committed:
reworked FPB .convertFile(); support for windows

snwagbata committed:
fixed bug with png file path on windows

snwagbata committed:
renamed .convertFile parameter

snwagbata committed:
Merge branch 'dev' of https://github.com/snwagbata/Parabeac-Core into dev

ivan-015 commented on lib/main.dart:
```diff
@@ -94,6 +94,11 @@ ${parser.usage}
     designType = 'figma';
   }
 
+  if (projectName == 'test') {
```
Hey @snwagbata I really like the idea of checking for invalid names before running parabeac-core. However, I do not think we should do this in `main.dart`, since in the future we could add more invalid names. I suggest adding maybe a separate class to detect for valid/invalid names would be a lot better, what do you think? 🤔

snwagbata replied:
That sounds like a great idea! Where would be the ideal location for having the class?

ivan-015 replied:
Currently verifying this with the team! I'll get back to you with an answer soon!

ivan-015 replied:
Hi @snwagbata, apologies for the delay. I think the best place for this file would be in the `lib/controllers` directory. Perhaps it could be named something like `project_name_verifier.dart` or something along those lines. Let me know what you think or if you have any other ideas!

snwagbata replied:
Hey @ivan-015, I went with `project_name_validator.dart` for clarity. Also, I discovered Dart's linter package could handle package name validation, so I went with that. Nevertheless, I also implemented a way for adding invalid names in the future.

ivan-015 commented on parabeac.dart:
```diff
@@ -71,33 +71,35 @@ ${parser.usage}
-    '${Directory.current.path}/pb-scripts/install.sh',
-  ]);
+  if (_os != 'WIN') {
+    var install = await Process.start('bash', [
```
@snwagbata I believe both `install.sh` and `merge-plugins.sh` should be run regardless of platform. `install.sh` can be omitted only if Figma is being run. Feel free to ask questions if you have any! Also, apologies for the delay in the PR review!

snwagbata replied:
The problem is that `.sh` files do not run on PowerShell and CMD on windows. I've had success getting them to run using git bash though

ivan-015 replied:
I see. I think the problem is that if we don't merge plugins we will not be able to detect when people want eggs and if someone in windows wants to use Sketch they will not be able to run the Sketch Asset Converter. Do you happen to know if there is an alternative solution to running these scripts on Windows? I usually run parabeac-core on unix so I'm not sure what the best approach would be for this 🤔

snwagbata replied:
The alternative would be to use the Windows Subsystem for Linux. I attempted rewriting `install.sh` to a batch file a few days ago but it seems the Sketch Asset Converter has a dependency that only works on unix

snwagbata committed:
implemented project name validator

snwagbata committed:
prevent install.sh if idFigmaInput

ivan-015 commented:
Hi @snwagbata, I've been reviewing your changes these last couple of days. Now I'm trying to test this on Mac on my end and another member of the team is trying to get this tested on Windows. However, the `pub get` command fails on both OS because of linter and build dependencies. 
Does this happen to you by chance? If so, what changes did you make to the pubspec.yaml file to get this to work?

Thank you again for your PR!

snwagbata commented:
Hey @ivan-015, I added `linter: ^0.1.127` to the pubspec.yaml file.

ivan-015 commented:
Hi @snwagbata, I've been trying to do a bit of research on this, but haven't been able to run your file yet. Attached is the message I get when I run `pub get`. Is it possible you could send me a screenshot of the output when you run `pub get`? Additionally, could you please tell me your flutter version? Thank you for your help.

Finally, it seems there are conflicts in `main.dart`. Unfortunately I am not able to fix these from my end since I cannot push directly to your branch. 

![image](https://user-images.githubusercontent.com/42812006/104083030-4cf81c80-51f8-11eb-94c0-d23f3f2bc15b.png)

snwagbata committed:
Merge branch 'dev' of https://github.com/Parabeac/Parabeac-Core into dev

snwagbata commented:
I was initially using a beta version of flutter, but I switched to stable v1.22.5 and it still ran with no problems. Instead of `pub get`, try running `pub upgrade` to see if the error persists.

ivan-015 commented:
Here is a screenshot of the error I receive 
![image](https://user-images.githubusercontent.com/42812006/104108540-c2103400-5282-11eb-8498-6793943ff408.png)

Another one of the members of Parabeac had also previously tried to resolve these dependencies with no luck, I find it quite strange that the command is working on your end 🤔. As mentioned in the error screen, the problem is that linter is incompatible with `json_serializable`

snwagbata committed:
resolve dependencies

snwagbata commented:
@ivan-015 I just pushed a commit to change `json_serializable`'s version as the recent version has bumped the analyzer's version to `>=0.39.0 <0.42.0`. Try running `pub get` on it. If it fails again, then I might have to include a dependency override in the `pubspec.yaml` file

snwagbata commented:
Hey @ivan-015, it seems I misread your earlier message. I using dart `2.10.4 (stable)`

ivan-015 commented:
Here's is the output I get when running pub get. Seemingly there are still issues 🤔 
![image](https://user-images.githubusercontent.com/42812006/104137153-10433700-5358-11eb-8524-4bab8212bd36.png)

snwagbata commented:
@ivan-015 I think I might have to reimplement the validator without the linter. Will push a fix sometime this weekend

snwagbata committed:
Extracted specific methods from linter packages for project name validating

snwagbata committed:
Extracted specific methods from linter packages for project name validating

snwagbata commented:
Hey @ivan-015, I just updated the project name validator. I extracted some methods used in the linter package for the validator and added a new package in the `pubspec.yaml`

ivan-015 commented:
I will be closing this Pull Request since the recent refactors on Parabeac-core have moved the sketch and figma controllers to PBDL.
